postgres/datadog_checks/postgres/util.py

```diff
@@ -293,14 +293,17 @@ def get_schema_field(descriptors):
     'name': 'pg_replication_slots',
     'query': """
     SELECT
-        slot_name,
-        slot_type,
-        CASE WHEN temporary THEN 'temporary' ELSE 'permanent' END,
-        CASE WHEN active THEN 'active' ELSE 'inactive' END,
-        CASE WHEN xmin IS NULL THEN NULL ELSE age(xmin) END,
-        pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn),
-        pg_wal_lsn_diff(pg_current_wal_lsn(), confirmed_flush_lsn)
-    FROM pg_replication_slots;
+        a.slot_name,
+        a.slot_type,
+        CASE WHEN a.temporary THEN 'temporary' ELSE 'permanent' END,
+        CASE WHEN a.active THEN 'active' ELSE 'inactive' END,
+        CASE WHEN xmin IS NULL THEN NULL ELSE age(a.xmin) END,
+        pg_wal_lsn_diff(cur.lsn, a.restart_lsn),
+        pg_wal_lsn_diff(cur.lsn, a.confirmed_flush_lsn)
+    FROM pg_replication_slots as a
+    CROSS JOIN (
+        SELECT
+        CASE WHEN pg_is_in_recovery() THEN pg_last_wal_receive_lsn() ELSE pg_current_wal_lsn() END) as cur(lsn);
```

Comment:
While the cross join looks correct and fine, it is a bit harder to reason about for non db experts. What would you think of this version without cross join? Unless we really need to protect ourselves from changes to `pg_last_wal_receive_lsn()` mid-query, this should be the same as what you've got.
 
```suggestion
        pg_wal_lsn_diff(CASE WHEN pg_is_in_recovery() THEN pg_last_wal_receive_lsn() ELSE pg_current_wal_lsn() END, a.restart_lsn),
        pg_wal_lsn_diff(CASE WHEN pg_is_in_recovery() THEN pg_last_wal_receive_lsn() ELSE pg_current_wal_lsn() END, a.confirmed_flush_lsn)
    FROM pg_replication_slots
```

Reply:
Both code snippets achieve the same result. You're right, I included the cross join because of [changes that could occur to pg_last_wal_receive_lsn() mid-query] but your adjustment is simple and organized.

Reply:
Thanks for the update, @boluwaji-deriv. I think this is almost good to merge but CI is currently failing because of formatting/lines too long. Do you mind doing one more update to please the style checker? I think that `ddev test -fs postgres` might be able to automatically do this.

Reply:
Thanks. This has been fixed